### PR TITLE
Prepare rails app for mobile interface

### PR DESCRIPTION
**Pull request by Scharrre97**
* Add gem ‘rack-cors’, require: ‘rack/cors’
* Allow crud actions in application.rb
* Add rspec configurations

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -26,5 +26,11 @@ class Application < Rails::Application
       generate.routing_specs false
       generate.controller_specs false
     end
+    config.middleware.insert_before 0, Rack::Cors do
+      allow do
+        origins '*'
+        resource '*', headers: :any, methods: [:get, :post, :put, :delete]
```
Comment (AmberWilkie) [inline config/application.rb]: Are you planning to use `put` and `delete` methods?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -14,6 +16,18 @@
...
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
+  end
+  
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+end
```
Comment (AmberWilkie) [inline spec/rails_helper.rb]: What does all this stuff do? Are you getting these things from Cooper?

Comment (fottander) [reply spec/rails_helper.rb]: Yeah we got it from the cooper documentation. @tochman Can you have a look at it?

Comment (AmberWilkie) [reply spec/rails_helper.rb]: @tochman @diraulo  I don't know what all these things do - if they need them or not for this project. They copied this code from the Cooper challenge.

Comment (tochman) [reply spec/rails_helper.rb]: Not sure what it does tbh

Comment (tochman) [reply spec/rails_helper.rb]: I would remove it and reintroduce it if it's needed.

File: spec/support/shoulda_matcher.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Shoulda::Matchers.configure do |config|
+  config.integrate do |with|
+    with.test_framework :rspec
+    with.library :rails
+  end
+end
+
+RSpec.configure do |config|
+  config.include(Shoulda::Matchers::ActiveRecord, type: :model)
+end
```
Comment (tochman) [inline spec/support/shoulda_matcher.rb]: This is already part of `rails_helper`

File: Gemfile [hunk 4]
```diff
@@ -36,3 +36,6 @@ group :development do
 end
 
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
```
Comment (tochman) [inline Gemfile]: I did not notice this line befor. This should be removed.

File: Gemfile [hunk 5]
```diff
@@ -36,3 +36,6 @@ group :development do
...
+# Use Rack CORS for handling Cross-Origin Resource Sharing (CORS),
+# making cross-origin AJAX possible
```
Comment (tochman) [inline Gemfile]: No need to include comments into your Gemfile

File: Gemfile [hunk 6]
```diff
@@ -36,3 +36,6 @@ group :development do
...
+gem 'rack-cors', require: 'rack/cors'
```
Comment (tochman) [inline Gemfile]: Move this to the top group before development and test groups.

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -14,6 +16,16 @@
...
 
+RSpec.configure do |config|
+  config.expect_with :rspec do |expectations|
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
+  end
+end
```
Comment (tochman) [inline spec/rails_helper.rb]: No need for a new configure block. C'mon guys!

**Review by tochman — APPROVED** [review]
All good.

**Pull request merged**